lineage/states/stateCommon.py

```diff
@@ -105,3 +102,41 @@ def basic_censor(cell):
             if not cell.get_sister().isLeafBecauseTerminal():
                 cell.get_sister().left.observed = False
                 cell.get_sister().right.observed = False
+
+
+def negative_LL_atonce(x, uncens_obs, uncens_gammas, cens_obs, cens_gammas):
+    """ uses the negative_LL_atonce and passes the vector of scales and the shared shape parameter. """
+    outt = 0.0
+    for i in range(4):
+        outt += negative_LL_sep(x[1 + i], x[0], uncens_obs[i], uncens_gammas[i], cens_obs[i], cens_gammas[i])
+    return outt
+
+
+def gamma_estimator_atonce(gamma_obs, time_cen, gamas, x0=None):
```

Comment:
@Farnazmdi I think what will help more than anything else is to pass the parameters from the last iteration in as a starting point (`x0`). Now, the first iteration won't satisfy the bounds, so you could write a check in here to override the starting point if it isn't valid. We do this with the other estimator, and it means that the first iteration takes the same amount of time, but subsequent ones are much faster.

Reply:
Sorry, this had slipped out of my sight. I'm on it.